lib/datasource/repology/index.ts

```diff
@@ -0,0 +1,126 @@
+      repo: repoName,
+      name_type: pkgType,
+      target_page: 'api_v1_project',
+      noautoresolve: 'on',
+      name: pkgName,
+    }).toString();
+
+    const url = `https://repology.org/tools/project-by?${query}`;
+    const res = await got(url, { hostType: id, json: true });
+
+    return res.body.find(
+      (x) => x.repo.toLowerCase() === repoName.toLowerCase()
+    );
+  } catch (err) {
+    if (err.statusCode === 404) {
+      logger.debug(
+        { repoName, pkgName, pkgType },
+        'Repository or package not found on Repology'
+      );
+    } else if (err.statusCode === 403) {
+      logger.debug(
+        { repoName },
+        'Repology does not support tools/project-by lookups for repository'
+      );
+    } else {
+      logger.warn(
+        { repoName, pkgName, pkgType, err },
+        'Repology package lookup failed with unexpected error'
+      );
+      throw new DatasourceError(err);
+    }
+  }
+
+  return null;
+}
+
+async function getCachedPackage(
+  repoName: string,
+  pkgName: string
+): Promise<RepologyPackage> {
+  // Fetch previous result from cache if available
+  const cacheKey = `${repoName}/${pkgName}`;
+  const cachedResult = await globalCache.get<RepologyPackage>(
+    cacheNamespace,
+    cacheKey
+  );
+  // istanbul ignore if
+  if (cachedResult) {
+    return cachedResult;
+  }
+
+  // Attempt a binary package lookup and return if successfully
+  const binPkg = await queryPackage(repoName, pkgName, 'binname');
+  if (binPkg) {
+    await globalCache.set(cacheNamespace, cacheKey, binPkg, cacheMinutes);
+    return binPkg;
+  }
+
+  // Otherwise, attempt a source package lookup and return if successfully
+  const srcPkg = await queryPackage(repoName, pkgName, 'srcname');
+  if (srcPkg) {
+    await globalCache.set(cacheNamespace, cacheKey, srcPkg, cacheMinutes);
+    return srcPkg;
+  }
+
+  // No binary or source package was found on Repology
+  return null;
+}
+
+export async function getReleases({
+  lookupName,
+}: GetReleasesConfig): Promise<ReleaseResult | null> {
+  // Ensure lookup name contains both repository and package
+  const [repoName, pkgName] = lookupName.split('/', 2);
+  if (!repoName || !pkgName) {
+    throw new DatasourceError(
+      new Error(
+        'Repology lookup name must contain repository and package separated by slash (<repo>/<pkg>)'
+      )
+    );
+  }
+
+  // Attempt to resolve package version through Repology
+  logger.trace(`repology.getReleases(${repoName}, ${pkgName})`);
+  try {
+    // Attempt to retrieve (cached) package information from Repology
+    const pkg = await getCachedPackage(repoName, pkgName);
+    if (!pkg) {
+      return null;
+    }
+
+    // Always prefer origversion if available, otherwise default to version
+    // This is required as source packages usually have no origversion
+    const version = pkg.origversion ?? pkg.version;
+    return { releases: [{ version }] };
+  } catch (err) {
+    throw new DatasourceError(err);
```

Comment:
Check for datasource error and rethrow. otherwise log warning
```suggestion
    if (err instanceof DatasourceError) {
      throw err;
    }
    logger.warn(
        { lookupName, err },
        'Repology lookup failed with unexpected error'
      );
    throw new DatasourceError(err);
```

Reply:
Good catch! I've rearranged the error handling altogether, so `queryPackage` now returns the raw error in case something went wrong. This eliminated the need of additional logic branching within `getReleases` and looks cleaner as well.